### PR TITLE
Seats choosing

### PR DESCRIPTION
**Pull request by butterflyInACast**

### CODE REVIEW

File: src/app/modules/ticket-reservation/services/reservation.service.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import { WebSocketService } from 'app/websocket/webSocket.service';
+import { WS_EVENTS } from 'app/websocket/webSocket.events';
+import { takeUntil } from 'rxjs/operators';
+import { BlockedSeat } from '@shared/models/blockedSeat';
+import { User } from '@shared/models/user';
+
+@Injectable()
+export class ReservationService implements OnDestroy {
+  private blockedSeatsValues: BlockedSeat[] = [];
+  private sessionSubject: Subject<Session> = new Subject();
```
Comment (viento006) [inline src/app/modules/ticket-reservation/services/reservation.service.ts]: ` = new  Subject<Session> ();` ?

File: src/app/modules/ticket-reservation/components/services-cart/services-cart.component.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  onServiceAdd(serviceOrder: ServiceOrder) {
+    this.servicesService.addToCart(serviceOrder);
+  }
+
+  removeFromCart(serviceOrder: ServiceOrder) {
+    this.servicesService.removeFromCart(serviceOrder);
+  }
+
+  isInCart(serviceId: string) {
+    return this.serviceOrders.some(order => order.service._id === serviceId);
```
[the remark this answers is not in the document]
Comment (butterflyInACast) [reply src/app/modules/ticket-reservation/components/services-cart/services-cart.component.ts]: I use this method only in the template of services-cart, so I don't see the value in moving it to the service

File: src/app/modules/ticket-reservation/components/services-cart/services-cart.component.html [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      Now services are available now...
+    </li>
+  </ul>
+
+  <h2><span uk-icon="icon: cart; ratio: 1.2"></span> Your cart</h2>
+  <ul class="cart">
+    <li *ngFor="let serviceOrder of serviceOrders" class="item">
+      <div class="row">
+        <span clas="name">{{ serviceOrder.service.name }}</span>
+        <span clas="amount">{{ 'x' + serviceOrder.amount }}</span>
```
Comment (viento006) [inline src/app/modules/ticket-reservation/components/services-cart/services-cart.component.html]: x{{....}}

File: src/app/modules/ticket-reservation/components/services-cart/services-cart.component.html [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        <span clas="price">{{
+          serviceOrder.amount * serviceOrder.service.price + ' BYN'
```
Comment (viento006) [inline src/app/modules/ticket-reservation/components/services-cart/services-cart.component.html]: {{serviceOrder.amount * serviceOrder.service.price}} BYN

File: src/app/modules/ticket-reservation/services/services.service.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import { distinctUntilChanged, map } from 'rxjs/operators';
+import { Service } from '@shared/models/service';
+import { Observable, BehaviorSubject } from 'rxjs';
+import { Response } from '@shared/models/response';
+import { ServiceOrder } from '@shared/models/serviceOrder';
+import { StorageKeys } from '@shared/enums/storageKeys';
+import { Cart } from '@shared/models/cart';
+
+@Injectable({
+  providedIn: 'root'
```
Comment (viento006) [inline src/app/modules/ticket-reservation/services/services.service.ts]: Shopuldn't it be provided in ticket-reservation module?

**Pull request merged**